src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/CodeFlowOperation.cs

```diff
@@ -1,43 +1,179 @@
+
+        // We create a temporary branch at the current checkout
+        // We flow the changes into another temporary branch
+        // Later we merge tmpBranch2 into tmpBranch1
+        // Then we look at the diff and stage that from the original repo checkout
+        // This way user only sees the staged files
+        string tmpTargetBranch = "darc/tmp/" + Guid.NewGuid().ToString();
+        string tmpHeadBranch = "darc/tmp/" + Guid.NewGuid().ToString();
+
+        try
+        {
+            await targetRepo.CreateBranchAsync(tmpTargetBranch, true);
+            await targetRepo.CreateBranchAsync(tmpHeadBranch, true);
+
+            Build build = new(-1, DateTimeOffset.Now, 0, false, false, currentFlow.SourceSha, [], [], [], [])
+            {
+                GitHubRepository = sourceRepo.Path,
+            };
+
+            bool hasChanges;
+
+            try
+            {
+                hasChanges = await _codeFlower.FlowCodeAsync(
+                    lastFlow,
+                    currentFlow,
+                    productRepo,
+                    mapping,
+                    build,
+                    excludedAssets: [],
+                    tmpTargetBranch,
+                    tmpHeadBranch,
+                    discardPatches: true,
+                    headBranchExisted: false,
+                    cancellationToken);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to flow changes between the VMR and {repo}", mappingName);
+
+                throw new InvalidSynchronizationException(
+                    "Failed to flow changes on top of the checked out commit - possibly due to conflicts. " +
+                    $"Changes are ready in the {tmpHeadBranch} branch (based on older version of the repo).");
+            }
+
+            if (!hasChanges)
+            {
+                _logger.LogInformation("No changes to flow between the VMR and {repo}.", mapping.Name);
+                await targetRepo.CheckoutAsync(currentTargetRepoBranch);
+                return;
+            }
+
+            await StageChangesFromBranch(targetRepo, mappingName, currentTargetRepoBranch, tmpHeadBranch, cancellationToken);
+        }
+        catch
+        {
+            await targetRepo.ResetWorkingTree();
+            await targetRepo.CheckoutAsync(currentTargetRepoBranch);
+            throw;
+        }
+        finally
+        {
+            _logger.LogInformation("Cleaning up...");
+
+            try
+            {
+                await targetRepo.DeleteBranchAsync(tmpTargetBranch);
+            }
+            catch
+            {
+            }
+            try
+            {
+                await targetRepo.DeleteBranchAsync(tmpHeadBranch);
+            }
+            catch
+            {
+            }
+
+            await sourceRepo.CheckoutAsync(currentSourceRepoBranch);
+        }
+
+        _logger.LogInformation("Changes staged in {repoPath}", targetRepo.Path);
```

Comment:
Nit: this method is pretty big and has a lot of _visual_ nesting. The logic flow is understandable, just that there are quite a few chained try/catches that make it seem bigger and more complex on the first read

Reply:
I will see if I can take some sections out into named methods to reduce the nesting. Though I quite like you can see the end to end procedure in one go.

Reply:
>I quite like you can see the end to end procedure in one go

Yeah, that's pretty nice. Hence why its just a nitpick, up to you if you want to change it 🙇